theseus/optimizer/sparse_linearization.py

```diff
@@ -38,7 +38,7 @@ def __init__(
         cost_function_row_block_starts = []  # where data start for this row block
         cost_function_stride = []  # total jacobian cols
 
-        for _, cost_function in enumerate(self.objective):
+        for _, cost_function in enumerate(self.objective._vectorized_iter()):
```

Comment:
I am wondering if it is clearer to have something like objective.get_iterator(), which returns the regular one or the vectorized one based on whether the objective is vectorize or not.

Reply:
What you are describing is the current behavior, but I like your name better. But do note that I deliberately named this as private, since I think I'm not sure it's safe for the user to have access to this iterator. If they update variables and try to loop over vectorized cost functions, they will get stale data. This is only safe to use inside linearization classes.

Reply:
That makes sense. Let's go with the more general name in that case for the private iterator.

I also realized there may be cases where the user still wants to access the default iterator through __iter__ even if they have vectorized, so we should still have a separate function like above.

Reply:
Renamed to `_get_iterator()`